### PR TITLE
API for RTC issue #14

### PR DESCRIPTION
**Pull request by brentward**

### CODE REVIEW

File: src/ria/main.c [hunk 1]
```diff
@@ -58,6 +59,7 @@ static void init()
     mou_init();
     rom_init();
     led_init();
+    rt_clock_init();
```
Comment (rumbledethumps) [inline src/ria/main.c:62]:
```suggestion
    rtc_init();
```

File: src/ria/api/rtc.c [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        uint32_t sec = 1;
+        int32_t nsec = 0;
+        if (!api_push_int32(&nsec) ||
+            !api_push_uint32(&sec))
+            return api_return_errno(API_EUNKNOWN);
+        api_sync_xstack();
+        return api_return_ax(0);
+    }
+    else
+        return api_return_errno(API_EUNKNOWN);
```
Comment (rumbledethumps) [inline src/ria/api/rtc.c:54]:
```suggestion
        return api_return_errno(API_EINVAL);
```

File: src/ria/api/rtc.c [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+void rtc_api_get_res(void)
+{
+    uint8_t clock_id = API_A;
+    if (clock_id == RIA_CLOCK_REALTIME)
+    {
+        uint32_t sec = 1;
+        int32_t nsec = 0;
+        if (!api_push_int32(&nsec) ||
+            !api_push_uint32(&sec))
+            return api_return_errno(API_EUNKNOWN);
```
Comment (rumbledethumps) [inline src/ria/api/rtc.c:49]:
```suggestion
            return api_return_errno(API_EINVAL);
```

File: src/ria/api/rtc.c [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            .tm_hour = rtc_info.hour,
+            .tm_min = rtc_info.min,
+            .tm_sec = rtc_info.sec,
+            .tm_isdst = -1,
+        };
+        time_t rawtime = mktime(&timeinfo);
+        int32_t rawtime_nsec = 0;
+        if (!api_push_int32(&rawtime_nsec) ||
+            !api_push_uint32((uint32_t *)&rawtime))
+            return api_return_errno(API_EUNKNOWN);
```
Comment (rumbledethumps) [inline src/ria/api/rtc.c:78]:
```suggestion
            return api_return_errno(API_EINVAL);
```

File: src/ria/api/rtc.c [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+            .sec = timeinfo.tm_sec,
+        };
+        if (!rtc_set_datetime(&rtc_info))
+            return api_return_errno(API_EUNKNOWN);
+
+        else
+            return api_return_ax(0);
+    }
+    else
+        return api_return_errno(API_EUNKNOWN);
```
Comment (rumbledethumps) [inline src/ria/api/rtc.c:113]:
```suggestion
        return api_return_errno(API_EINVAL);
```

File: src/ria/api/rtc.c [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+void rtc_api_set_time(void)
+{
+    uint8_t clock_id = API_A;
+    if (clock_id == RIA_CLOCK_REALTIME)
+    {
+        time_t rawtime;
+        int32_t rawtime_nsec;
+        if (!api_pop_uint32((uint32_t *)&rawtime) ||
+            !api_pop_int32_end(&rawtime_nsec))
+            return api_return_errno(API_EUNKNOWN);
```
Comment (rumbledethumps) [inline src/ria/api/rtc.c:95]:
```suggestion
            return api_return_errno(API_EINVAL);
```

File: src/ria/main.c [hunk 7]
```diff
@@ -192,6 +194,15 @@ bool main_api(uint8_t operation)
     case 0x1A:
         std_api_lseek();
         break;
+    case 0x10:
```
Comment (rumbledethumps) [inline src/ria/main.c:197]: Please keep in order.

File: src/ria/main.c (hunk 1 above)
Comment (brentward) [reply src/ria/main.c:62]: The `#include "hardware/rtc.h"` contains `void rtc_init(void);` so I don't think we can use that name. Is there a way to do it or a different name we can use?

Comment (rumbledethumps) [reply src/ria/main.c:62]: This and the ntp code in one driver wouldn't be too much. So ntp.c? Unless you got a better TLA?

**Comment by brentward** [comment]
Sorry for the extra push. I didn't add the CMakeLists because it has the version number and a 0 code page and missed that I had updated the target sources in it.

**Comment by rumbledethumps** [comment]
My bad. I shouldn't have accepted this without a license. Start here:
```
/*
 * Copyright (c) 2023 Brent Ward
 * Copyright (c) 2023 Rumbledethumps
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */
```

**Pull request merged**